src/core/Song.cpp

```diff
@@ -1133,6 +1134,13 @@ void Song::loadProject( const QString & fileName )
 
 	emit projectLoaded();
 
+	if( isCancelled() )
```

Comment:
I think this should be right after the `while` loop. But then LMMS will crash if you cancel loading a project which has single-streamed instruments. However, I have a fix for this as I mentioned before, and I'm going to push this today.

Anyway it works nice for projects without such instruments. :+1: